ble/GapAdvertisingData.h

```diff
@@ -345,6 +431,27 @@ class GapAdvertisingData
         return (uint16_t)_appearance;
     }
 
+    /**
+     * Search advertisement data for field.
+     * Returns pointer to field if found, NULL otherwise.
+     */
+    uint8_t* findField(DataType_t type) {
```

Comment:
if it is useful outside, then it is fine to leave it as public. But I would then have a second version of this function which returns a const uint8_t * and operates on a const object; i.e. `const uint8_t* findField(DataType_t type) const {`. The two versions can share code; the non-const version can call into the const version.  The const version could be public. The non-const equivalent should be private.